sql/opt/exec/execbuilder: TestExecBuild failed

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2534524&tab=buildLog) on [master@86323a52b12250d5a8895754bd8de935e18dd473](https://github.com/cockroachdb/cockroach/commits/86323a52b12250d5a8895754bd8de935e18dd473):

```
=== RUN   TestExecBuild
    test_log_scope.go:72: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild413303791
    test_log_scope.go:73: use -show-logs to present logs inline
=== CONT  TestExecBuild
    logic.go:3166: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild413303791
--- FAIL: TestExecBuild (82.47s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (0.00s)
=== RUN   TestExecBuild/local/upsert
=== PAUSE TestExecBuild/local/upsert
=== CONT  TestExecBuild/local/upsert
=== CONT  TestExecBuild/local/upsert
    logic.go:2156: 
         
        testdata/upsert:364: set tracing=off;
        SELECT operation, message FROM [SHOW KV TRACE FOR SESSION]
         WHERE operation != 'dist sender send'
        expected:
            materializer                          Scan /Table/57/1/2{-/#}
            materializer                          fetched: /kv/primary/2/v -> /3
            flow                                  Put /Table/57/1/2/0 -> /TUPLE/2:2:Int/2
            flow                                  Del /Table/57/2/3/0
            flow                                  CPut /Table/57/2/2/0 -> /BYTES/0x8a (expecting does not exist)
            kv.DistSender: sending partial batch  r35: sending batch 1 Put, 1 EndTxn to (n1,s1):1
            exec stmt                             execution failed after 0 rows: duplicate key value violates unique constraint "woo"\nDETAIL: Key (v)=(2) already exists\.
            
        but found (query options: "") :
            materializer                          Scan /Table/57/1/2{-/#}
            materializer                          fetched: /kv/primary/2/v -> /3
            flow                                  Put /Table/57/1/2/0 -> /TUPLE/2:2:Int/2
            flow                                  Del /Table/57/2/3/0
            flow                                  CPut /Table/57/2/2/0 -> /BYTES/0x8a (expecting does not exist)
            kv.DistSender: sending partial batch  r35: sending batch 1 Put, 1 EndTxn to (n1,s1):1
            exec stmt                             execution failed after 0 rows: duplicate key value violates unique constraint "woo"
--- done: testdata/upsert with config local: 21 tests, 1 failures
E201221 23:47:44.648565 122 sql/logictest/logic.go:3398  testdata/upsert:377: too many errors encountered, skipping the rest of the input
    logic.go:2956: 
        testdata/upsert:377: error while processing
    logic.go:2956: testdata/upsert:377: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/local/upsert (5.23s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2535073&tab=buildLog) on [master@1a6ee6ebd9625f428568203205d0ae44e2ca1963](https://github.com/cockroachdb/cockroach/commits/1a6ee6ebd9625f428568203205d0ae44e2ca1963):

```
testdata/upsert:364: set tracing=off;
SELECT operation, message FROM [SHOW KV TRACE FOR SESSION]
 WHERE operation != 'dist sender send'
expected:
    materializer                          Scan /Table/57/1/2{-/#}
    materializer                          fetched: /kv/primary/2/v -> /3
    flow                                  Put /Table/57/1/2/0 -> /TUPLE/2:2:Int/2
    flow                                  Del /Table/57/2/3/0
    flow                                  CPut /Table/57/2/2/0 -> /BYTES/0x8a (expecting does not exist)
    kv.DistSender: sending partial batch  r35: sending batch 1 Put, 1 EndTxn to (n1,s1):1
    exec stmt                             execution failed after 0 rows: duplicate key value violates unique constraint "woo"\nDETAIL: Key (v)=(2) already exists\.
    
but found (query options: "") :
    materializer                          Scan /Table/57/1/2{-/#}
    materializer                          fetched: /kv/primary/2/v -> /3
    flow                                  Put /Table/57/1/2/0 -> /TUPLE/2:2:Int/2
    flow                                  Del /Table/57/2/3/0
    flow                                  CPut /Table/57/2/2/0 -> /BYTES/0x8a (expecting does not exist)
    kv.DistSender: sending partial batch  r35: sending batch 1 Put, 1 EndTxn to (n1,s1):1
    exec stmt                             execution failed after 0 rows: duplicate key value violates unique constraint "woo"
    logic.go:2156: 
         
        testdata/upsert:364: set tracing=off;
        SELECT operation, message FROM [SHOW KV TRACE FOR SESSION]
         WHERE operation != 'dist sender send'
        expected:
            materializer                          Scan /Table/57/1/2{-/#}
            materializer                          fetched: /kv/primary/2/v -> /3
            flow                                  Put /Table/57/1/2/0 -> /TUPLE/2:2:Int/2
            flow                                  Del /Table/57/2/3/0
            flow                                  CPut /Table/57/2/2/0 -> /BYTES/0x8a (expecting does not exist)
            kv.DistSender: sending partial batch  r35: sending batch 1 Put, 1 EndTxn to (n1,s1):1
            exec stmt                             execution failed after 0 rows: duplicate key value violates unique constraint "woo"\nDETAIL: Key (v)=(2) already exists\.
            
        but found (query options: "") :
            materializer                          Scan /Table/57/1/2{-/#}
            materializer                          fetched: /kv/primary/2/v -> /3
            flow                                  Put /Table/57/1/2/0 -> /TUPLE/2:2:Int/2
            flow                                  Del /Table/57/2/3/0
            flow                                  CPut /Table/57/2/2/0 -> /BYTES/0x8a (expecting does not exist)
            kv.DistSender: sending partial batch  r35: sending batch 1 Put, 1 EndTxn to (n1,s1):1
            exec stmt                             execution failed after 0 rows: duplicate key value violates unique constraint "woo"
--- done: testdata/upsert with config local: 21 tests, 1 failures
E201222 06:12:28.314351 184672 sql/logictest/logic.go:3398  testdata/upsert:377: too many errors encountered, skipping the rest of the input
    logic.go:2956: 
        testdata/upsert:377: error while processing
    logic.go:2956: testdata/upsert:377: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/local/upsert (2.34s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (216.35s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2535061&tab=buildLog) on [master@1a6ee6ebd9625f428568203205d0ae44e2ca1963](https://github.com/cockroachdb/cockroach/commits/1a6ee6ebd9625f428568203205d0ae44e2ca1963):

```
=== RUN   TestExecBuild
    test_log_scope.go:72: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild630351559
    test_log_scope.go:73: use -show-logs to present logs inline
=== CONT  TestExecBuild
    logic.go:3166: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild630351559
--- FAIL: TestExecBuild (53.47s)
=== RUN   TestExecBuild/local/upsert
=== PAUSE TestExecBuild/local/upsert
=== CONT  TestExecBuild/local/upsert
=== CONT  TestExecBuild/local/upsert
    logic.go:2156: 
         
        testdata/upsert:364: set tracing=off;
        SELECT operation, message FROM [SHOW KV TRACE FOR SESSION]
         WHERE operation != 'dist sender send'
        expected:
            materializer                          Scan /Table/57/1/2{-/#}
            materializer                          fetched: /kv/primary/2/v -> /3
            flow                                  Put /Table/57/1/2/0 -> /TUPLE/2:2:Int/2
            flow                                  Del /Table/57/2/3/0
            flow                                  CPut /Table/57/2/2/0 -> /BYTES/0x8a (expecting does not exist)
            kv.DistSender: sending partial batch  r35: sending batch 1 Put, 1 EndTxn to (n1,s1):1
            exec stmt                             execution failed after 0 rows: duplicate key value violates unique constraint "woo"\nDETAIL: Key (v)=(2) already exists\.
            
        but found (query options: "") :
            materializer                          Scan /Table/57/1/2{-/#}
            materializer                          fetched: /kv/primary/2/v -> /3
            flow                                  Put /Table/57/1/2/0 -> /TUPLE/2:2:Int/2
            flow                                  Del /Table/57/2/3/0
            flow                                  CPut /Table/57/2/2/0 -> /BYTES/0x8a (expecting does not exist)
            kv.DistSender: sending partial batch  r35: sending batch 1 Put, 1 EndTxn to (n1,s1):1
            exec stmt                             execution failed after 0 rows: duplicate key value violates unique constraint "woo"
--- done: testdata/upsert with config local: 21 tests, 1 failures
E201222 06:15:49.915131 128 sql/logictest/logic.go:3398  testdata/upsert:377: too many errors encountered, skipping the rest of the input
    logic.go:2956: 
        testdata/upsert:377: error while processing
    logic.go:2956: testdata/upsert:377: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/local/upsert (3.75s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (0.00s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/opt/exec/execbuilder/testdata/upsert
+++ b/pkg/sql/opt/exec/execbuilder/testdata/upsert
@@ -372,7 +372,7 @@ flow                                  Put /Table/57/1/2/0 -> /TUPLE/2:2:Int/2
 flow                                  Del /Table/57/2/3/0
 flow                                  CPut /Table/57/2/2/0 -> /BYTES/0x8a (expecting does not exist)
 kv.DistSender: sending partial batch  r35: sending batch 1 Put, 1 EndTxn to (n1,s1):1
-exec stmt                             execution failed after 0 rows: duplicate key value violates unique constraint "woo"\nDETAIL: Key (v)=(2) already exists\.
+exec stmt                             execution failed after 0 rows: duplicate key value violates unique constraint "woo"
 
 
 subtest regression_32473

